bin/pytorch_inference/Main.cc

```diff
@@ -168,52 +171,70 @@ template<std::size_t N>
 void writePrediction(const torch::Tensor& prediction,
                      const std::string& requestId,
                      std::uint64_t timeMs,
-                     ml::core::CRapidJsonConcurrentLineWriter& jsonWriter) {
+                     TRapidJsonLineWriter& jsonWriter) {
 
-    // creating the accessor will throw if the tensor does
-    // not have exactly N dimensions. Do this before writing
-    // any output so the error message isn't mingled with
-    // a partial result
+    // Creating the accessor will throw if the tensor does not have exactly
+    // N dimensions. Do this before writing any output so the error message
+    // isn't mingled with a partial result.
 
     if (prediction.dtype() == torch::kFloat32) {
         auto accessor = prediction.accessor<float, N>();
-
         writeInferenceResultOpening(requestId, timeMs, jsonWriter);
         writeInferenceResults(accessor, jsonWriter);
         writeInferenceResultClosing(jsonWriter);
 
     } else if (prediction.dtype() == torch::kFloat64) {
         auto accessor = prediction.accessor<double, N>();
-
         writeInferenceResultOpening(requestId, timeMs, jsonWriter);
         writeInferenceResults(accessor, jsonWriter);
         writeInferenceResultClosing(jsonWriter);
+
     } else {
         std::ostringstream ss;
-        ss << "cannot process result tensor of type [" << prediction.dtype() << "]";
+        ss << "Cannot process result tensor of type [" << prediction.dtype() << "]";
         writeError(requestId, ss.str(), jsonWriter);
     }
 }
 
+rapidjson::Document updateRequestId(const std::string& responseJson,
+                                    const std::string& requestId) {
+    rapidjson::Document response;
+    response.Parse(responseJson.c_str());
+    if (response.HasMember(RESULT)) {
```

Comment:
Agreed. It would be nice is we can make REQUEST_ID separate so we can decouple from writing inference response; we don't really want to even store it in the cache. I didn't rejig the result format at this stage, but we can discuss if we want to do this as part of this change.

Reply:
It requires a Java change for the parsing, it can easily be done in a later PR